tests/test_datasource.py

```diff
@@ -65,28 +75,45 @@ def test_csv_with_sep(path):
     ds = DataSource(path("0_0_sep.csv"), reader_kwargs={"sep": ","})
     assert ds.get_df().shape == (2, 1)
 
+    ds = DataSource(
+        path("0_0_sep.csv"), reader_kwargs={"preview": {"nrows": 1, "offset": 0}, "sep": ","}
+    )
+    assert ds.get_df().shape == (1, 1)
+
 
 def test_csv_with_encoding(path):
     """It should be able to detect the encoding if not set"""
     df = DataSource(path("latin_1.csv")).get_df()
     assert df.shape == (2, 7)
     assert "unité économique" in df.columns
 
+    df = DataSource(
+        path("latin_1.csv"), reader_kwargs={"preview": {"nrows": 1, "offset": 1}}
+    ).get_df()
+    assert df.shape == (1, 7)
+
 
 def test_csv_with_sep_and_encoding(path):
     """It should be able to detect everything"""
     ds = DataSource(path("latin_1_sep.csv"))
     assert ds.get_df().shape == (2, 7)
 
+    ds = DataSource(path("latin_1_sep.csv"), reader_kwargs={"preview": {"nrows": 1, "offset": 0}})
+    assert ds.get_df().shape == (1, 7)
+
 
 def test_read_pandas(path):
     """It should be able to detect everything with read_pandas shortcut"""
     assert read_pandas(path("latin_1_sep.csv")).shape == (2, 7)
+    assert read_pandas(path("latin_1_sep.csv"), preview={"nrows": 1, "offset": 1}).shape == (1, 7)
 
 
 def test_read_pandas_excel(path):
     """It should be able to detect everything with read_pandas shortcut"""
     assert read_pandas(path("0_2.xls"), keep_default_na=False).shape == (2, 2)
+    assert read_pandas(
+        path("0_2.xls"), keep_default_na=False, preview={"nrows": 1, "offset": 0}
+    ).shape == (1, 2)
```

Comment:
We should probably check the content to show nrows and offset work fine for all the tests

Reply:
👍🏾 Updated !